nwb_conversion_tools/datainterfaces/ecephys/cellexplorer/cellexplorerdatainterface.py

```diff
@@ -17,28 +24,51 @@ class CellExplorerSortingInterface(BaseSortingExtractorInterface):
-        if spikes_cellinfo_file_path.is_file():
-            cell_info = loadmat(spikes_cellinfo_file_path).get("spikes", np.empty(0))
-            cell_info_fields = cell_info.dtype.names
+
+        assert spikes_matfile_path.is_file(), f"The spikes_matfile_path ({spikes_matfile_path}) must exist!"
+
+        try:
+            spikes_mat = scipy.io.loadmat(file_name=str(spikes_matfile_path))
+            self.read_spikes_info_with_scipy = True
+        except NotImplementedError:
+            spikes_mat = hdf5storage.loadmat(file_name=str(spikes_matfile_path))
+            self.read_spikes_info_with_scipy = False
+
+        cell_info = spikes_mat.get("spikes", np.empty(0))
+        self.cell_info_fields = cell_info.dtype.names
+
+        if self.read_spikes_info_with_scipy:  # Logic for scipy
             unit_ids = self.sorting_extractor.get_unit_ids()
-            if "cluID" in cell_info_fields:
+            if "cluID" in self.cell_info_fields:
                 for unit_id, value in zip(unit_ids, [int(x) for x in cell_info["cluID"][0][0][0]]):
                     self.sorting_extractor.set_unit_property(unit_id=unit_id, property_name="clu_id", value=value)
-            if "shankID" in cell_info_fields:
+            if "shankID" in self.cell_info_fields:
                 for unit_id, value in zip(unit_ids, [f"Group{x}" for x in cell_info["shankID"][0][0][0]]):
                     self.sorting_extractor.set_unit_property(unit_id=unit_id, property_name="group_id", value=value)
-            if "region" in cell_info_fields:
+            if "region" in self.cell_info_fields:
                 for unit_id, value in zip(unit_ids, [str(x[0]) for x in cell_info["region"][0][0][0]]):
                     self.sorting_extractor.set_unit_property(unit_id=unit_id, property_name="location", value=value)
+        else:  # Logic for hdf5
+            unit_ids = self.sorting_extractor.get_unit_ids()
+            if "cluID" in self.cell_info_fields:
+                for unit_id, value in zip(unit_ids, [int(x) for x in cell_info["cluID"][0][0]]):
+                    self.sorting_extractor.set_unit_property(unit_id=unit_id, property_name="clu_id", value=value)
+            if "shankID" in self.cell_info_fields:
+                for unit_id, value in zip(unit_ids, [f"Group{x}" for x in cell_info["shankID"][0][0]]):
+                    self.sorting_extractor.set_unit_property(unit_id=unit_id, property_name="group_id", value=value)
+            if "region" in self.cell_info_fields:
+                for unit_id, value in zip(unit_ids, [str(x[0]) for x in cell_info["region"][0]][0]):
+                    self.sorting_extractor.set_unit_property(unit_id=unit_id, property_name="location", value=value)
 
+        # CellClass file
         celltype_mapping = {"pE": "excitatory", "pI": "inhibitory", "[]": "unclassified"}
         celltype_file_path = session_path / f"{session_id}.CellClass.cellinfo.mat"
```

Comment:
```suggestion
        celltype_mapping = {"pE": "excitatory", "pI": "inhibitory", "[]": "unclassified"}
        celltype_file_path = session_path / f"{session_id}.CellClass.cellinfo.mat"
```
Clear enough without added line/comment.